### DISCUSSION TITLE
Feedback needed: What are you using Titanoboa for? What would you like to see in the next release?

### DISCUSSION THREAD

**Discussion by mikub** [mikub/titanoboa#10] [General]
Before I jump the gun and decide what to put on the roadmap for the next Titanoboa releases, I thought I would like to hear from the community:
- are you using it? What are you using it for?
- what do you like? What you don't like?
- what is missing? What would you like to see in the next release(s)?
- or maybe you considered using Titanoboa, but eventually decided not to use it because ... ?

Feel free to leave comments, all and any feedback is appreciated!

**Comment by aadrian** [comment]
> what do you like?

The UI diagrams and Clojure :) .

> What you don't like?

There are no prices on the main site. This usually scares away many potential small customers.

> what is missing? What would you like to see in the next release(s)?

A few small ones _(in random order)_.
- more tasklets. These https://github.com/mikub/titanoboa-tasklets are not that many _(for most projects to adopt this solution)_
- formalize/spec and enforce tasklet definition so that the community can (and would be motivated) to easily contribute back many new tasklets.
- export/import workflows from the UI
- make the main website accessible.
- the GUI webapp should not use CDN
- FF browser console is full of warnings (and also errors)
- browser URL does not change with route changes.
- missing tooltips on many buttons/icons in the UI
- missing Undo/Redo (at least in the Workflow Editor).
- distribution needs a main directory (the latest archive contains none)
- control scripts for Windows too (there's only one for Windows)
- log messages seem to miss the timestamps
- offer "official" Docker images too
- publish the community release under https://github.com/mikub/titanoboa/releases automatically with GitHub Actions.
- notifications and mobile optimizations (and screenshots :) ), so that at least the progress and status of important jobs can be easily  followed from mobile.
 - since you also offer hosting, an online demo that can be tried with simple click (the app could get reset every few hours to avoid spam)

**Comment by mikub** [comment]
Thanks @aadrian, all good points! Appreciate the time you took to write them up. I might create a github issue/enhancement ticket for each of them. 
Feel free to submit any furher issues in future if anything comes up!

Few notes:

> export/import workflows from the UI

This is actually very simple - you just copy/paste EDN from UI's EDN tab. It is so simple I thought I did not need to add any import/export button... But maybe the ability is not self-explanatory and people might miss it - so I may reconsider adding the button...

> since you also offer hosting, an online demo that can be tried with simple click (the app could get reset every few hours to avoid spam)

This is actually done and almost ready to go - including the "reset every few hours" :) (just deciding between 2 or 3hrs :D ; you can actually see how it works at the beginning of the recent demo/video)

> formalize/spec and enforce tasklet definition so that the community can (and would be motivated) to easily contribute back many new tasklets.

Defo a good point, but I will need to think this through, since the main benefit of Titanoboa is that you can just code away and tweak anything you need on the fly (this was the main pain I was having with other similar products - you always have to customize, but it is usually not trivial to customize the predefined steps.) So not sure if by enforcing tasklet definition I would be able to keep this level of flexibility in the solution... 
But yes, definitely at least making an optional spec for each step/tasklet has crossed my mind few times already, the hard part of that will be probably specifying and visualisation of the spec (and spec checks) in the UI...

> FF browser console is full of warnings (and also errors)

Slowly have been cleaning them up, majority are react/reagent warnings/errors regarding some of the UI's elements' IDs.

> control scripts for Windows too

Yeah yeah I know, those windows ;)

**Comment by mikub** [comment]
Also got few few tips regarding Titanoboa's roadmap on reddit, here is the [link](https://www.reddit.com/r/Clojure/comments/i89rd8/titanoboa_feedback_needed_are_you_using_it_what/)
